### PR TITLE
Add constructor for windowing `nothing` boundary conditions

### LINKED ISSUE #3687: `view` fails on a `Field` indexed from `FieldTimeSeries`
It's a bug:

```julia
julia> using Oceananigans

julia> grid = RectilinearGrid(size=(3, 3), x=(0, 1), y=(0, 1), topology=(Periodic, Periodic, Flat))
3×3×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
├── Periodic x ∈ [0.0, 1.0) regularly spaced with Δx=0.333333
├── Periodic y ∈ [0.0, 1.0) regularly spaced with Δy=0.333333
└── Flat z

julia> ct = FieldTimeSeries{Center, Center, Center}(grid, (0, 1))
3×3×1×2 FieldTimeSeries{InMemory} located at (Center, Center, Center) on CPU
├── grid: 3×3×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
├── indices: (:, :, :)
├── time_indexing: Linear()
├── backend: InMemory()
└── data: 9×9×1×2 OffsetArray(::Array{Float64, 4}, -2:6, -2:6, 1:1, 1:2) with eltype Float64 with indices -2:6×-2:6×1:1×1:2
    └── max=0.0, min=0.0, mean=0.0

julia> view(ct[1], :, :, 1)
ERROR: MethodError: no method matching topology(::Tuple{Colon, Colon, UnitRange{Int64}})

Closest candidates are:
  topology(::Any, ::Any)
   @ Oceananigans ~/Projects/Oceananigans.jl/src/Grids/grid_utils.jl:59
  topology(::Oceananigans.Fields.AbstractField, Any...)
   @ Oceananigans ~/Projects/Oceananigans.jl/src/Fields/abstract_field.jl:47
  topology(::Oceananigans.Grids.AbstractGrid{FT, TX, TY, TZ}) where {FT, TX, TY, TZ}
   @ Oceananigans ~/Projects/Oceananigans.jl/src/Grids/grid_utils.jl:52

Stacktrace:
 [1] topology
   @ ~/Projects/Oceananigans.jl/src/Grids/grid_utils.jl:59 [inlined]
 [2] FieldBoundaryConditions(grid::Tuple{Colon, Colon, UnitRange{Int64}}, location::Nothing, indices::Tuple{Colon, Colon, Colon})
   @ Oceananigans.BoundaryConditions ~/Projects/Oceananigans.jl/src/BoundaryConditions/field_boundary_conditions.jl:135
 [3] view(f::Field{…}, i::Function, j::Function, k::Int64)
   @ Oceananigans.Fields ~/Projects/Oceananigans.jl/src/Fields/field.jl:326
 [4] top-level scope
   @ REPL[13]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

### PR DESCRIPTION
**Pull request by glwagner**
Resolves #3687

### CODE REVIEW

**Review by simone-silvestri — APPROVED** [review]

File: src/Grids/grid_utils.jl [hunk 1]
```diff
@@ -269,12 +272,9 @@ end
 
 # intersect_index_range(::Colon, ::Colon) = Colon()
 index_range_contains(range,   subset::UnitRange) = (first(subset) ∈ range) & (last(subset) ∈ range)
-index_range_contains(::Colon, subset::UnitRange) = true
+index_range_contains(::Colon, ::UnitRange)       = true
 index_range_contains(::Colon, ::Colon)           = true
-
-# Note: this choice means subset indices are defined on the whole grid.
-# Thus any UnitRange does not contain `:`.
```
Comment (glwagner) [inline src/Grids/grid_utils.jl:276]: In hindsight this was a bad choice

**Comment by navidcy** [comment]
main is already at 0.91.7 so perhaps this needs bump up?

**Comment by glwagner** [comment]
> main is already at 0.91.7 so perhaps this needs bump up?

it'll sneak in if we haven't registered yet

**Comment by glwagner** [comment]
oh nevermind we did

**Comment by glwagner** [comment]
its ok we don't need to tag

**Pull request merged**